solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -1352,20 +1333,33 @@ contract LiquidityProtection is ILiquidityProtection, Utils, Owned, ReentrancyGu
     // utility to get the reserve balances
     function converterReserveBalances(
         IConverter converter,
-        IERC20 reserveToken1,
-        IERC20 reserveToken2
+        IReserveToken reserveToken1,
+        IReserveToken reserveToken2
     ) private view returns (uint256, uint256) {
         return (converter.getConnectorBalance(reserveToken1), converter.getConnectorBalance(reserveToken2));
     }
 
     // utility to get the other reserve
-    function converterOtherReserve(IConverter converter, IERC20 thisReserve) private view returns (IERC20) {
-        IERC20 otherReserve = converter.connectorTokens(0);
+    function converterOtherReserve(IConverter converter, IReserveToken thisReserve)
+        private
+        view
+        returns (IReserveToken)
+    {
+        IReserveToken otherReserve = converter.connectorTokens(0);
```

Comment:
What do you mean?

Reply:
We are potentially calling external function `connectorTokens` twice here.
Instead, since the StandardPoolConverter contract implements function `reserveTokens` which returns both reserve token addresses, we can do a single external call to this function.

Reply:
We can make this change as long as the interface is common for LPV1 as well, but we can do it after its deprecation

Reply:
Why? LP contract uses only SP by definition.

Reply:
It's using `IConverter` (by definition), which doesn't have `reserveTokens`, since legacy converters don't implement it, so as long as they aren't deprecated (which is in the plans) - we can't use this function

Reply:
Well, since LP contract uses only SP by definition, we can cast the given converter to an interface which declares `reserveTokens`, and be sure that it will run correctly. But this is not critical.

Reply:
The spec needs to (and will) be implemented in the future. Doing this now is ugly and premature.

Reply:
Let's stay productive and focus on the PR itself